compiler/ir/ir.psi2ir/src/org/jetbrains/kotlin/psi2ir/KotlinUtils.kt

```diff
@@ -54,11 +54,15 @@ fun ResolvedCall<*>.isValueArgumentReorderingRequired(): Boolean {
 }
 
 fun KtSecondaryConstructor.isConstructorDelegatingToSuper(bindingContext: BindingContext): Boolean {
-    val delegatingResolvedCall = getDelegationCall().getResolvedCall(bindingContext) ?: return false
     val constructorDescriptor = bindingContext.get(BindingContext.CONSTRUCTOR, this) ?: return false
-    val ownerClassDescriptor = constructorDescriptor.containingDeclaration
-    val targetClassDescriptor = delegatingResolvedCall.resultingDescriptor.containingDeclaration
-    return targetClassDescriptor != ownerClassDescriptor
+    val delegatingResolvedCall = getDelegationCall().getResolvedCall(bindingContext)
```

Comment:
Looks like K/N also could be affected

Reply:
Oh, you're right. Could you please run the K/N tests on this branch? I still haven't managed to set this up properly.

Reply:
"Composite Kotlin/Native sanity test" build is successful

Reply:
@sfs Looks like this change also requires irText test  (@dnpetrov please take a look)

Reply:
@sfs Looks ok to me.
Please, add corresponding psi2ir test (test data located at compiler/testData/ir/irText).

Reply:
I've added a test based on `secondaryConstructors/enums.kt`